js/ui/camera.js

```diff
@@ -352,21 +351,67 @@ util.extend(Camera.prototype, /** @lends Map.prototype */{
 
         bounds = LatLngBounds.convert(bounds);
 
-        var offset = Point.convert(options.offset),
-            tr = this.transform,
-            nw = tr.project(bounds.getNorthWest()),
+        var offset = Point.convert(options.offset);
+
+        var tr = this.transform;
+
+        var nw = tr.project(bounds.getNorthWest()),
             se = tr.project(bounds.getSouthEast()),
-            size = se.sub(nw),
-            scaleX = (tr.width - options.padding * 2 - Math.abs(offset.x) * 2) / size.x,
-            scaleY = (tr.height - options.padding * 2 - Math.abs(offset.y) * 2) / size.y;
+            ne = tr.project(bounds.getNorthEast()),
+            sw = tr.project(bounds.getSouthWest());
+
+        var segment = [ nw, sw, se, ne ];
+
+        var nePixel = {
+            x: -Infinity,
+            y: -Infinity
+        };
+        var swPixel = {
+            x: Infinity,
+            y: Infinity
+        };
+
+        for (var i in segment) {
```

Comment:
Better to use `forEach()` or a plain ol’ for loop in this case. In C++, the for…in syntax was cleaner than the alternative, but there isn’t much to be gained here.